include/gz/math/MovingWindowFilter.hh

```diff
@@ -90,6 +90,17 @@ namespace gz
     using MovingWindowFilterVector3f = MovingWindowFilter<Vector3f>;
     using MovingWindowFilterVector3d = MovingWindowFilter<Vector3d>;
   }
+
+
+/// The template explicit instantiations that does not use fundamental types
+/// (like Vector*) needs to be placed in this header file.
+template class GZ_MATH_VISIBLE MovingWindowFilter<int>;
```

Comment:
I'm not sure if putting explicit instantiations in a header is the right thing to do (see https://stackoverflow.com/questions/5864401/does-explicit-template-instantiation-go-in-cpp-or-header-file). If every user of this header instantiates these classes, we'd have ODR issues. Also, we'd lose the benefit of explicit instantiation since every translation unit will be instantiating the templates.